crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -119,8 +117,15 @@ impl CallContractCommand {
 	}
 
 	/// Checks if the contract has been built; if not, builds it.
+	/// If the path is a contract artifact file, skips the build process
 	async fn ensure_contract_built(&self, cli: &mut impl Cli) -> Result<()> {
-		// Check if build exists in the specified "Contract build directory"
+		// The path is expected to be set. If it is not, exit early without attempting to build the contract.
+		let Some(path) = self.path.as_deref() else { return Ok(()) };
+		// Check if the path is a file or the build exists in the specified "Contract build
+		// directory"
+		if !path.is_dir() || has_contract_been_built(self.path.as_deref()) {
```

Comment:
```suggestion
		if path.is_file() || has_contract_been_built(self.path.as_deref()) {
```
Improves readability.